### DISCUSSION TITLE
[The Rest of Your Life] Listing 1 low noise

### DISCUSSION THREAD

**Discussion by genmeblog** [RayTracing/raytracing.github.io#1016] [Debugging]
The image generated in listing 1 (or listing 3 in version 4.x) contains very low noise. My result however is quite noisy.

In book v.4, the only difference is `samples_per_pixel` set to 100 (instead of 64 from beginning of the book). It doesn't create such smooth image as [here](https://github.com/RayTracing/raytracing.github.io/blob/book1-review/images/img-3.03-cornell-refactor1.jpg).
In book v.3, `pdf` can have negative values and the result is very noisy.

Book v4 version from my code:
![ch06a](https://user-images.githubusercontent.com/38646601/166818465-73381d93-04b1-493a-a128-49a504e56058.jpg)

Is it bug in the book or in my code?

(samples_per_pixel set to 1000 gives the result similar to this from the book)

**Comment by hollasch** [comment]
Converted to issue #1018
